sos/upload/profiles/redhat.py

```diff
@@ -0,0 +1,246 @@
+            self.ui_log.info("No case id provided, uploading to SFTP")
+
+    def prompt_for_upload_password(self):
+        # With OIDC we don't ask for user/pass anymore
+        if self.commons['cmdlineopts'].upload_pass:
+            self.ui_log.info(
+                _("The option --upload-pass has been deprecated in favour"
+                  " of device authorization in RHEL")
+            )
+
+    def get_upload_url(self):
+        rh_case_api = "/support/v1/cases/%s/attachments"
+        try:
+            if self.upload_url:
+                return self.upload_url
+            if self.commons['cmdlineopts'].upload_url:
+                return self.commons['cmdlineopts'].upload_url
+            if self.commons['cmdlineopts'].upload_protocol == 'sftp':
+                return self.RH_SFTP_HOST
+            if not self.commons['cmdlineopts'].case_id:
+                return self.RH_SFTP_HOST
+
+        except Exception as e:
+            self.ui_log.info(
+                "There was a problem while setting the "
+                f"remote upload target:  {e}"
+            )
+        return self.RH_API_HOST + rh_case_api\
+            % self.commons['cmdlineopts'].case_id
+
+    def _get_upload_https_auth(self):
+        str_auth = f"Bearer {self._device_token}"
+        return {'Authorization': str_auth}
+
+    def _upload_https_post(self, archive, verify=True):
+        """If upload_https() needs to use requests.post(), use this method.
+
+        Policies should override this method instead of the base upload_https()
+
+        :param archive:     The open archive file object
+        """
+        files = {
+            'file': (archive.name.split('/')[-1], archive,
+                     self._get_upload_headers())
+        }
+        # Get the access token at this point. With this,
+        # we cover the cases where report generation takes
+        # longer than the token timeout
+        RHELAuth = DeviceAuthorizationClass(
+                self.client_identifier_url,
+                self.token_endpoint
+            )
+        self._device_token = RHELAuth.get_access_token()
+        self.ui_log.info("Device authorized correctly. Uploading file to "
+                         f"{self.get_upload_url_string()}")
+        return requests.post(self.get_upload_url(), files=files,
+                             headers=self._get_upload_https_auth(),
+                             verify=verify, timeout=TIMEOUT_DEFAULT)
+
+    def _get_upload_headers(self):
+        if self.get_upload_url().startswith(self.RH_API_HOST):
+            return {'isPrivate': 'false', 'cache-control': 'no-cache'}
+        return {}
+
+    def get_upload_url_string(self):
+        if self.get_upload_url().startswith(self.RH_API_HOST):
+            return "Red Hat Customer Portal"
+        if self.get_upload_url().startswith(self.RH_SFTP_HOST):
+            return "Red Hat Secure FTP"
+        return self._get_obfuscated_upload_url(self.upload_url)
```

Comment:
Related to the above comment for `get_upload_url()` and #3891:

```suggestion
        if self.get_upload_url().startswith(self.RH_SFTP_HOST):
            return "Red Hat Secure FTP"
        if self.get_upload_url().startswith('s3://'):
            return f"{self.get_upload_url()} on endpoint {self.get_upload_s3_endpoint()}"
        return self._get_obfuscated_upload_url(self.upload_url)
```

Reply:
Looks good to me but lets wait for other's feedback.